apps/nextjs/src/components/layout/header/navigation-bar.tsx

```diff
@@ -0,0 +1,46 @@
+"use client";
+
+import Link from "next/link";
+import { usePathname } from "next/navigation";
+import { Group, NavLink } from "@mantine/core";
+
+import { clientApi } from "@homarr/api/client";
+
+export const NavigationBar = () => {
+  const { data: boards = [] } = clientApi.board.getAllBoards.useQuery(undefined, {
+    refetchOnMount: false,
+    refetchOnWindowFocus: false,
+  });
+  const pathname = usePathname();
+
+  // Filter boards that should be shown in navigation
+  const navigationBoards = boards.filter((board) => board.showInNavigation);
```

Comment:
Create a new route that returns only those from the navigation bar

Reply:
Also I would suggest, that we fetch this data on the server and pass it down to this component, then it does not need to make a roundtrip. THen we define it as initialData so we still can invalidate the values when adding one